src/lib/features/segment/admin-segment.e2e.test.ts

```diff
@@ -204,14 +204,8 @@ test('should create segments', async () => {
     expect(segments.map((s) => s.name)).toEqual(['a', 'b', 'c']);
 });
 
-test('should return 400 for non numeeric segment ID', async () => {
-    const { body } = await app.request
-        .get(`${SEGMENTS_BASE_PATH}/stringName`)
-        .expect(400);
-    expect(body).toMatchObject({
-        message:
-            'Request validation failed: your request body or params contain invalid data: ID should be an integer',
-    });
+test('should return 400 for non numeric segment ID', async () => {
+    await app.request.get(`${SEGMENTS_BASE_PATH}/stringName`).expect(400);
```

Comment:
I think it is enough to check to respond with status code 400 instead of checking both status code and error message.